src/commands/command-lib/mod/streammessage.cmdlib.ts

```diff
@@ -0,0 +1,158 @@
+                        .setRequired(true)
+                )
+                .addRoleOption((option) =>
+                    option.setName("role").setDescription("Select a role")
+                )
+                .addStringOption((option) =>
+                    option
+                        .setName("template")
+                        .setDescription("Select a template")
+                        .addChoices({
+                            name: "stream", value: "stream"
+                        }, {
+                            name:"cancellation", value: "cancellation"})
+                )
+        );
+        this._loaded = true;
+        return true;
+    }
+
+    public async runSubCommand(interaction: Discord.CommandInteraction) {
+        await interaction.deferReply({
+            ephemeral: true,
+        });
+
+        //Get Channel List
+        const channels = await interaction.client.guilds.fetch(global.app.config.mainServer).then(guild => guild.channels.fetch())
+        // check if there is a channel that includes "birthdays" in it's name
+        let channel;
+        if (!channels.some((channel) => /live-(streams|notifications|notifs|announcements)/i.test(channel.name) && channel.type == Discord.ChannelType.GuildText)) {
+
+            if (!channels.some((channel) => /announcements/i.test(channel.name) && channel.type == Discord.ChannelType.GuildText)) {
+                interaction.editReply({content: "No channel found with the name `live-streams`, `live-notifications`, `live-notifs`, `live-announcements`, or `announcements`"});
+                return;
+            } else {
+                channel = channels.find((channel) => /announcements/i.test(channel.name) && channel.type == Discord.ChannelType.GuildText);
+            }
+            
+        } else {
+            channel = channels.find((channel) => /live-(streams|notifications|notifs|announcements)/i.test(channel.name) && channel.type == Discord.ChannelType.GuildText);
+        }
+
+        const userinfo = await storage.findOne("user", {
+            id: interaction.user.id,
+        });
+        const usersTimezone = userinfo?.approximatedTimezone ?? "Europe/Berlin";
+
+        //Get options from command
+        const template = (interaction.options as Discord.CommandInteractionOptionResolver).getString("template");
+        const role = (
+            interaction.options as Discord.CommandInteractionOptionResolver
+        ).getRole("role") ?? "";
+        const message = (
+            interaction.options as Discord.CommandInteractionOptionResolver
+        ).getString("message");
+        const time = (
+            interaction.options as Discord.CommandInteractionOptionResolver
+        ).getString("time");
+
+        //Check if time is a valid time
+
+        if (template === "cancellation") {
+            await interaction.editReply({
+                content: `Ok sent to ${channel}`,
```

Comment:
I think an even better message could be:
```suggestion
                content: `Your message has been sent to {channel}` // Add a preview of the message if possible?
```